pkg/relayer/session/sessiontree.go

```diff
@@ -0,0 +1,192 @@
+package session
+
+import (
+	"crypto/sha256"
+	"os"
+	"path/filepath"
+	sessiontypes "pocket/x/session/types"
+	"sync"
+
+	"github.com/pokt-network/smt"
+)
+
+var _ SessionTree = (*sessionTree)(nil)
+
+// sessionTree is an implementation of the SessionTree interface.
+type sessionTree struct {
+	// sessionMu is a mutex used to protect sessionTree operations from concurrent access.
+	sessionMu *sync.Mutex
+
+	// session is the Session corresponding to the SMST (Sparse Merkle State Tree).
+	session *sessiontypes.Session
+
+	// tree is the SMST (Sparse Merkle State Tree) corresponding the session.
+	tree *smt.SMST
+
+	// claimedRoot is the root hash of the SMST needed for submitting the claim.
+	// If it holds a non-nil value, it means that the SMST has been flushed, committed to disk
+	// and no more updates can be made to it. It also indicates that a proof could be generated
```

Comment:
Does it being not nil indicate the proof can be generated? If so please clarify this in the comment